controllers/utils/utils.go

```diff
@@ -167,7 +168,48 @@ func VerifyIfCapabilityIsEnabled(ctx context.Context, cli client.Client, capabil
 	}
 
 	return false, nil
+}
+
+// GetIstioControlPlaneName return the name of the Istio Control Plane and the mesh namespace.
+// It will first try to read the environment variables, if not found will then try to read the DSCI
+// If the required value is not available in the DSCI, it will return the default values
+func GetIstioControlPlaneName(ctx context.Context, cli client.Client) (istioControlPlane string, meshNamespace string) {
+	// first try toi retrieve it from the envs, it should be available through the service-mesh-refs ConfigMap
+	istioControlPlane = os.Getenv("CONTROL_PLANE_NAME")
+	meshNamespace = os.Getenv("MESH_NAMESPACE")
 
+	if len(istioControlPlane) == 0 || len(meshNamespace) == 0 {
+		objectList, _ := getDSCIObject(ctx, cli)
```

Comment:
So, the error is not handled. This is not wrong, _per se_.

However, if there is an error, then `objectList.Items` will be empty. In this case, the code inside the `for` loop will never run. So, `istioControlPlane`  and/or `meshNamespace` variables will remain unset, and empty values will be returned.

Looks like a check is needed to guarantee that, at least, default values returned.

Reply:
indeed, nice catch.